feat(protocol-designer, step-generation): update mix for inPlace, expand tipstate for pipettes

ncdiehl11 opened the pull request:
## Overview

This PR wires up `aspirate/dispenseInPlace` commands to mix. It also expands tip state temporal properties for pipettes to include not only if a tip is attached, but what the tiprack URI of the attached tip is, so that we can check safe pipette movements in `moveToWell` commands through checking robot state at that command.

In `mix` compound command creator, we now emit:
- `moveToWell`
- `aspirateInPlace`
- `dispenseInPlace`

rather than:
- `aspirate`
- `dispense`

Also, I wire up pipette movement safety checks in `moveToWell`, requiring nozzle configuration to be passed. This has a large footprint, since I needed to update each implementation of `moveToWell` in step generation command creators.

## Test Plan and Hands on Testing

- smoke test by creating a mix step and saving, verifying that no timeline errors occur unexpectedly
- check out migration fixtures to make sure it all looks correct

## Changelog



## Review requests



## Risk assessment

ncdiehl11 committed:
feat(protocol-designer,-step-generation): update mix for inPlace, expand tipstate for pipettes

This PR wires up `aspirate/dispenseInPlace` commands to mix. It also expands tip state temporal properties for pipettes to include not only if a tip is attached, but what the tiprack URI of the attached tip is, so that we can check safe pipette movements in `moveToWell` commands through checking robot state at that command.

ncdiehl11 committed:
revert unintended change

jerader commented on step-generation/src/__tests__/transfer.test.ts:
```diff
@@ -129,7 +132,7 @@ describe('pick up tip if no tip on pipette', () => {
     })
   })
-  it('...once, drop tip in waste chute', () => {
+  it.only('...once, drop tip in waste chute', () => {
```
```suggestion
  it('...once, drop tip in waste chute', () => {
```

jerader commented on step-generation/src/types.ts:
```diff
@@ -637,7 +637,10 @@ export interface TimelineFrame {
-      [pipetteId: string]: boolean // true if pipette has tip(s)
+      [pipetteId: string]: {
+        hasTip: boolean
+        attachedTipURI: string | null
```
slight nit, this is actually the tiprack uri that the tip is a part of right? i think the naming is a bit confusing since it seems like the tip itself has a URI

```suggestion
        tiprackURI: string | null
```

ncdiehl11 committed:
refactor(protocol-designer,-step-generation): use robotState rather than explicit tiprack/nozzles props

This PR refactors the implementation of getIsSafePipetteMovement to pull nozzle configuration and attached tip URI from the robot state, rather than requiring additional explicit prop passing (no new information)

ncdiehl11 committed:
wire up tip check update in dropTipInPlace

jerader reviewed (approved):
code is a lot cleaner with this latest refactor. Smoke tested a bit and it works as expected - i also did some 96-channel testing and the pipette collision error came up when expected. nice!

ddcc4 commented on step-generation/src/utils/safePipetteMovements.ts:
```diff
@@ -291,27 +298,32 @@ export const getIsSafePipetteMovement = (
+  const tiprackURI = tipState.pipettes[pipetteId]?.tiprackURI
+  const tiprackEntityId =
+    tiprackURI != null
+      ? Object.keys(labwareEntities).find(lwKey => lwKey.includes(tiprackURI))
```
Hey @ncdiehl11 @jerader, `tipState.pipettes[pipetteId].tiprackURI` might be a misnomer ...

I think `tiprackURI` is not a URI, but a labware ID. It looks like something like `"b26cc212-0434-4d09-9148-f5de13773d4a:opentrons/opentrons_96_filtertiprack_200ul/1"`.

In which case, `lwKey.includes(tiprackURI)` here doesn't make sense. You don't need `includes()` if `tiprackURI` is the entire labware ID, right?
